hal/src/samd21/usb/bus.rs

```diff
@@ -218,15 +168,15 @@ impl BufferAllocator {
         let start_addr = unsafe { start_addr.offset(offset as isize) };
 
         if start_addr >= buf_end {
-            return Err(UsbError::SizeOverflow);
+            return Err(UsbError::EndpointMemoryOverflow);
         }
 
         let end_addr = unsafe { start_addr.offset(size as isize) };
         if end_addr > buf_end {
-            return Err(UsbError::SizeOverflow);
+            return Err(UsbError::EndpointMemoryOverflow);
         }
 
-        self.next_buf = unsafe { end_addr.offset_from(self.buffers.as_ptr()) as u16 };
+        self.next_buf = unsafe { end_addr.sub(self.buffers.as_ptr() as usize) as u16 };
```

Comment:
This is also copied from the samd51 example and avoids using an unstable feature